scripts/diagnostic_gen.py

```diff
@@ -280,7 +286,7 @@ def createallheader(path, diags):
 
 def createdocs(outDir, inpath, slangBin, diags, groups):
     inf = open(inpath)
-    curropt = None
+    curropt = [None, None, None, None, None]
```

Comment:
Why this change?

Reply:
This was our more strict python type-checker. It really did not like the fact that we are assigning array elements to a None type. It did not figure out that the array was initialized in one of the branches before it was assigned to. 

I agree that your code made sense, and it is correct for the way the input files are structured, but the static checker does not have access to the inputs, and so it was throwing errors. This, arguably more verbose solution, quiets it down.

Reply:
This is fine, I don't care too much one way or the other. I was just curious.